core/client/fs/src/main/java/alluxio/client/block/stream/BlockInStream.java

```diff
@@ -229,20 +139,14 @@ public boolean isShortCircuit() {
    * Creates an instance of {@link BlockInStream}.
    *
    * @param inputStream the packet inputstream
-   * @param blockWorkerClient the block worker client
-   * @param closer the closer registered with closable resources open so far
+   * @param workerNetAddress the worker network address
    * @param options the options
    */
-  protected BlockInStream(PacketInStream inputStream, BlockWorkerClient blockWorkerClient,
-      Closer closer, InStreamOptions options) {
+  protected BlockInStream(PacketInStream inputStream, WorkerNetAddress workerNetAddress,
+      InStreamOptions options) {
     super(inputStream);
-
     mInputStream = inputStream;
-    mBlockWorkerClient = blockWorkerClient;
-
-    mCloser = closer;
-    mCloser.register(mInputStream);
-    mLocal = blockWorkerClient.getWorkerNetAddress().getHost()
-        .equals(NetworkAddressUtils.getClientHostName());
+    mAddress = workerNetAddress;
+    mLocal = workerNetAddress.getHost().equals(NetworkAddressUtils.getClientHostName());
```

Comment:
this `addr.equals(NetworkAddressUtils.getClientHostName())` has been used multiple times, I think it worths making a util method for this purpose.